news/_posts/2020-11-30-from-legacy-to-future-architecture-connecting-the-dots.md

```diff
@@ -0,0 +1,308 @@
+
+We don't know exactly how the Front Office API will be implemented yet; it could be REST, it could be GraphQL... all bets are off. But what we do know is that it's a completely new component that sits at application layer (the topmost layer in the backend architecture described above). Because of that, its initial implementation won't be burdened by the Core's SemVer restrictions – it doesn't require us to wait for a major release. It could even be released as a composer package to allow for fast, iterative experimentation before it's fully merged into the Core. We can't wait to start working on it. 
+
+### Modernize front end development
+
+With the Front Office API in place, we will be able to start working a modern reference implementation for the Front Office, based on this new API and built with VueJS. However, building a modern web app is a complex matter, and building an extensible one is **even harder**. As this essay dives farther into the future, it's difficult for me to outline it in detail.
+
+That said, one thing is certain for me: such an app must be based on [reusable components][ui-kit], data stores and [event-driven state management][flux-data-flow]. They are the building blocks for a Software Development Kit (SDK) that core developers, module developers and integrators alike will be able to use to build PrestaShop and customize it through modules. Once more, I think it would be unwise to move the Front Office to VueJS without building this SDK first, and it's the reason why we stopped building fully VueJS-based Back Office pages in the first place.
+
+#### What about the Back Office?
+
+Once the Front-end SDK has been designed, it will be possible to move the Back Office into the VueJS / API world as well. Thanks to the Core API, building a Back Office API that powers VueJS pages will be a piece of cake, because everything will be already in place: Back Office API controllers will output JSON instead of HTML pages, but will still consume the same Commands and Queries as their current counterparts.
+
+## Final notes
+
+We don't yet know the exact order in which these ideas will be implemented, or for that matter, if all of them will be done precisely as I envision them today, or done at all. However, the intent of this series of articles isn't just to communicate "this is what we are going to do", and be done with it. These articles are a way put our analysis and our vision into words and share them with the community. They are meant to trigger a discussion so that we can find the best possible path to a common objective, a desirable future that we can look forward to building together: **making PrestaShop the best open source e-commerce platform there is.**
```

Comment:
```suggestion
We don't yet know the exact order in which these ideas will be implemented, or for that matter, if all of them will be done precisely as I envision them today, or done at all. However, the intent of this series of articles isn't just to communicate "this is what we are going to do", and be done with it. These articles are a way put our analysis and our vision into words and share them with the community. They are meant to trigger a discussion so that we can find the best possible path to a common objective, a desirable future that we can look forward to building together: **making PrestaShop the best open source e-commerce platform there is**.
```